src/Common/src/Common.Security/PemConfigureCertificateOptions.cs

```diff
@@ -12,7 +12,7 @@
 
 namespace Steeltoe.Common.Security;
 
-public class PemConfigureCertificateOptions : IConfigureNamedOptions<CertificateOptions>
+public sealed class PemConfigureCertificateOptions : IConfigureNamedOptions<CertificateOptions>
```

Comment:
The logic for reading PEM certificates can be removed, because the framework is now capable of reading PEM certificates, see https://stackoverflow.com/questions/50227580/create-x509certificate2-from-pem-file-in-net-core/66066129#66066129.

It can be simplified by exposing just two properties in `IConfiguration`:
- CertificateFilePath
- PrivateKeyFilePath

And have Steeltoe intelligently do "the right thing" to load the `X509Certificate2` into options. The simplest would be (from `IConfigureNamedOptions`):
```c#
public void Configure(CertificateOptions options)
{
    Configure(Options.DefaultName, options);
}

public void Configure(string? name, CertificateOptions options)
{
    // Do not overwrite when explicitly set from code.
    if (options.Certificate == null)
    {
        string? certificateFilePath = _configuration.GetValue<string>(
            GetConfigurationKey(name, "CertificateFilePath"));
        string? privateKeyFilePath = _configuration.GetValue<string>(
            GetConfigurationKey(name, "PrivateKeyFilePath"));

        if (certificateFilePath != null)
        {
            options.Certificate = privateKeyFilePath != null
                ? X509Certificate2.CreateFromPemFile(certificateFilePath, privateKeyFilePath)
                : new X509Certificate2(certificateFilePath);
        }
    }
}

private static string GetConfigurationKey(string? optionName, string propertyName)
{
    return string.IsNullOrEmpty(optionName)
        ? string.Join(ConfigurationPath.KeyDelimiter, CertificateOptions.ConfigurationKeyPrefix, propertyName)
        : string.Join(ConfigurationPath.KeyDelimiter, CertificateOptions.ConfigurationKeyPrefix, optionName, propertyName);
}
```
Which assumes PEM when both paths are provided. We can make it more advanced by looking at the file extension and/or probing into the first bytes of the file, it this turns out to be insufficient. There's a list of common formats and where the private key is stored at https://serverfault.com/questions/9708/what-is-a-pem-file-and-how-does-it-differ-from-other-openssl-generated-key-file/9717#9717. I've tested the above with the certificate files from tests, which properly imports the private key.

Reply:
Yeah, I already have changes related to reading the certificates pending in my other branch, but I'll bring those here now and merge with your suggestion